go/vt/vtgate/vtgateconn/vtgateconn.go

```diff
@@ -55,6 +54,19 @@ func (conn *VTGateConn) Session(targetString string, options *querypb.ExecuteOpt
 	}
 }
 
+// SessionPb returns the underlying proto session.
+func (sn *VTGateSession) SessionPb() *vtgatepb.Session {
+	return sn.session
+}
+
+// SessionFromPb returns a VTGateSession based on the provided proto session.
+func (conn *VTGateConn) SessionFromPb(sn *vtgatepb.Session) *VTGateSession {
+	return &VTGateSession{
+		session: sn,
+		impl:    conn.impl,
+	}
+}
+
```

Comment:
I'd like to get @sougou / @systay to comment on whether we are comfortable adding these.

Reply:
> I'd like to get @sougou / @systay to comment on whether we are comfortable adding these.

Feels like it's a necessary hack. I don't have an objection.